src/NUnitFramework/framework/TestContext.cs

```diff
@@ -295,6 +295,16 @@ public static void AddFormatter(ValueFormatterFactory formatterFactory)
         }
 
         /// <summary>
+        /// Attach a file to the current test result
+        /// </summary>
+        /// <param name="filepath">Relative or absolute filepath to attachment</param>
+        /// <param name="description">Optional description of attachment</param>
+        public static void AddTestAttachment(string filepath, string description = null)
+        {
+            TestExecutionContext.CurrentContext.AddTestAttachment(filepath, description);
```

Comment:
Why the extra level of indirection? In other code, TestContext just accesses `TestExecutionContext.CurrentContext.CurrentResult`  directly.